src/main/scala/cromwell/engine/backend/sge/SgeBackend.scala

```diff
@@ -0,0 +1,147 @@
+    val rc: Int = process.exitValue()
+    Vector(qdelStdoutWriter, qdelStderrWriter) foreach { _.flushAndClose() }
+    val (_, rcWriter) = backendCall.rc.fileAndWriter
+    rcWriter.writeWithNewline("143")
+    rcWriter.flushAndClose()
+    logger.debug(s"qdel $sgeJobId (rc=$rc)")
+  }
+
+  /**
+   * Writes the script file containing the user's command from the WDL as well
+   * as some extra shell code for monitoring jobs
+   */
+  private def writeScript(backendCall: BackendCall, instantiatedCommand: String) = {
+    val (_, scriptWriter) = backendCall.script.fileAndWriter
+    scriptWriter.writeWithNewline("#!/bin/sh")
+    scriptWriter.writeWithNewline(instantiatedCommand)
+    scriptWriter.writeWithNewline("echo $? > rc")
+    scriptWriter.flushAndClose()
+  }
+
+  /**
+   * Launches the qsub command, returns a tuple: (rc, Option(sge_job_id))
+   */
+  private def launchQsub(backendCall: BackendCall): (Int, Option[Int]) = {
+    val tag = makeTag(backendCall)
+    val sgeJobName = s"cromwell_${backendCall.workflowDescriptor.shortId}_${backendCall.call.name}"
+    val argv = Seq("qsub", "-N", sgeJobName, "-V", "-b", "n", "-wd", backendCall.callRootPath.toAbsolutePath, "-o", backendCall.stdout.getFileName, "-e", backendCall.stderr.getFileName, backendCall.script.toAbsolutePath).map(_.toString)
+    val backendCommandString = argv.map(s => "\""+s+"\"").mkString(" ")
+    logger.info(s"$tag backend command: $backendCommandString")
+
+    val (qsubStdoutFile, qsubStdoutWriter) = backendCall.callRootPath.resolve("qsub.stdout").fileAndWriter
+    val (qsubStderrFile, qsubStderrWriter) = backendCall.callRootPath.resolve("qsub.stderr").fileAndWriter
+    val process = argv.run(ProcessLogger(qsubStdoutWriter writeWithNewline, qsubStderrWriter writeWithNewline))
+    val rc: Int = process.exitValue()
+    Vector(qsubStdoutWriter, qsubStderrWriter) foreach { _.flushAndClose() }
+    val jobId = "^Your job (\\d+)".r.findFirstMatchIn(qsubStdoutFile.slurp).map(_.group(1).toInt)
+    logger.info(s"$tag qsub rc=$rc, job ID=${jobId.getOrElse("NONE")}")
+    (rc, jobId)
+  }
+
+  /**
+   * This waits for a given SGE job to finish.  When finished, it post-processes the job
+   * and returns the outputs for the call
+   */
+  private def pollForSgeJobCompletionThenPostProcess(backendCall: BackendCall, sgeJobId: Int): Try[CallOutputs] = {
+    val tag = makeTag(backendCall)
+    val abortFunction = killSgeJob(backendCall, sgeJobId)
+    val waitUntilCompleteFunction = waitUntilComplete(backendCall)
+    backendCall.callAbortRegistrationFunction.register(AbortFunction(abortFunction))
+    val jobRc = waitUntilCompleteFunction
+    logger.info(s"$tag SGE job completed (rc=$jobRc)")
+    (jobRc, backendCall.stderr.toFile.length) match {
+      case (r, _) if r != 0 =>
+        Failure(new Throwable(s"$tag SGE job failed because of non-zero return code: $r"))
+      case (r, _) if r == 143 =>
```

Comment:
Cases are matched in order.  The `r != 0` case will match for `r == 143` so this is unreachable.